tests/spec/s-array.js

```diff
@@ -1596,4 +1596,439 @@ describe('Array', function () {
             expect(result).toEqual(['2', '3']);
         });
     });
+
+    describe('#sort()', function () {
+        describe('usage', function () {
+            it('requires a function or undefined as first argument', function () {
+                var actual = [];
+                expect(actual.sort()).toBe(actual);
+                expect(actual.sort(undefined)).toBe(actual);
+                expect(actual.sort(function () { return 0; })).toBe(actual);
+            });
+
+            it('requires a non-function or non-undefined to throw a `TypeError`', function () {
+                expect(function () {
+                    try {
+                        [].sort(null);
+                    } catch (e) {
+                        expect(e).toEqual(jasmine.any(TypeError));
+                        throw e;
+                    }
+                }).toThrow();
+
+                expect(function () {
+                    try {
+                        [].sort(1);
+                    } catch (e) {
+                        expect(e).toEqual(jasmine.any(TypeError));
+                        throw e;
+                    }
+                }).toThrow();
+
+                expect(function () {
+                    try {
+                        [].sort('');
+                    } catch (e) {
+                        expect(e).toEqual(jasmine.any(TypeError));
+                        throw e;
+                    }
+                }).toThrow();
+
+                expect(function () {
+                    try {
+                        [].sort(true);
+                    } catch (e) {
+                        expect(e).toEqual(jasmine.any(TypeError));
+                        throw e;
+                    }
+                }).toThrow();
+
+                expect(function () {
+                    try {
+                        [].sort({});
+                    } catch (e) {
+                        expect(e).toEqual(jasmine.any(TypeError));
```

Comment:
instead of explicitly doing this assertion, you can do `.toThrow(TypeError)` and remove a lot of this boilerplate - ideally each of these assertions is on a single line.